patches/minecraft/net/minecraft/entity/item/EntityMinecart.java.patch

```diff
@@ -254,11 +288,22 @@
                              {
                                  p_70108_1_.field_70159_w *= 0.20000000298023224D;
                                  p_70108_1_.field_70179_y *= 0.20000000298023224D;
-@@ -1015,6 +1047,230 @@
+@@ -1015,6 +1030,274 @@
          this.func_184212_Q().func_187227_b(field_184270_f, Boolean.valueOf(p_94096_1_));
      }
  
 +    /* =================================== FORGE START ===========================================*/
++
++    /**
++     * The bitwise mask indicating the minecart is boosting, as if on a powered golden rail.
++     */
++    public static final int BOOSTING_MASK = 0x1;
++
++    /**
++     * The bitwise mask indicating the minecart is slowing, as if on a un-powered golden rail.
++     */
++    public static final int SLOWING_MASK = 0x2;
```

Comment:
maybe have an enum and use enumset? I always find raw int flags error-prone

Reply:
If you suggest this, I'd prefer modifiable boolean pairs

Reply:
bit-masked flags aren't so bad and there are places where they make sense (not so sure about here). However, hey Java8! At least use `0b0010` to make clear that it's bit-masked...

Reply:
I think the javadoc itself can clarify things

Reply:
Then you might as well write them decimally, considering 1 and 2 are the same in the hexadecimal as in decimal system. I'm in favour of using binary literals here as well to stress the *bit*wise nature of the flags.

As Henry suggests, I'd also use leading zeroes to indicate how many flags are possible/in active use.

Reply:
will do.